catalog_tools/io/parser.py

```diff
@@ -0,0 +1,233 @@
+import xml.sax
+from datetime import datetime
+
+
+def _get_realvalue(key: str, value: str) -> dict:
+    real_values = {'value': '',
+                   'uncertainty': '_uncertainty',
+                   'lowerUncertainty': '_lowerUncertainty',
+                   'upperUncertainty': '_upperUncertainty',
+                   'confidenceLevel': '_confidenceLevel'}
+    return {f'{key}{k}': f'{value}{v}' for k, v in real_values.items()}
+
+
+EVENT_MAPPINGS = {'publicID': 'eventid',
+                  'type': 'event_type'}
+
+ORIGIN_MAPPINGS = {
+    **_get_realvalue('origintime', 'time'),
+    **_get_realvalue('originlatitude', 'latitude'),
+    **_get_realvalue('originlongitude', 'longitude'),
+    **_get_realvalue('origindepth', 'depth')
+}
+
+MAGNITUDE_MAPPINGS = {
+    **_get_realvalue('magnitudemag', 'magnitude'),
+    'magnitudetype': 'magnitude_type',
+    'magnitudeevaluationMode': 'evaluationMode',
+}
+
+DUMMY_MAGNITUDE = {
+    'magnitudemagvalue': None,
+    'magnitudetype': None,
+    'magnitudeevaluationMode': None}
+
+DUMMY_ORIGIN = {
+    'origintimevalue': None,
+    'originlatitudevalue': None,
+    'originlongitudevalue': None,
+    'origindepthvalue': None
+}
+
+
+def _get_preferred_magnitude(magnitudes: list, id: str | None) \
+        -> tuple[dict, list]:
+    preferred = next((m for m in magnitudes if id
+                     == m['magnitudepublicID']), DUMMY_MAGNITUDE)
+
+    magnitudes = [m for m in magnitudes if m['magnitudetype']
+                  != preferred['magnitudetype'] or id == m['magnitudepublicID']]
+
+    return preferred, magnitudes
+
+
+def _get_preferred_origin(origins: list, id: str):
```

Comment:
Not sure if a doctoring is always needed, but for some of them I'm not really sure what they do..

Reply:
renamed it, but if the functions are very small, internal, and mostly for code readability, I don't bother (as long as the name is descriptive)